core/service/manager.go

```diff
@@ -171,7 +182,18 @@ func (manager *Manager) Kill() error {
 
 // Stop stops the service.
 func (manager *Manager) Stop(id ID) error {
-	return manager.servicePool.Stop(id)
+	instance := manager.servicePool.Instance(id)
+	if instance == nil {
+		return errors.New("service not found")
+	}
+
+	err := manager.servicePool.Stop(id)
+	if err != nil {
+		return err
+	}
+
+	manager.eventPublisher.Publish(StopTopic, instance)
```

Comment:
I expect all session stop logic to live in `servicePool.Stop()`. No point to do searching twice, it's already done there

Reply:
At first I've done all cleanup directly, but then I've added this event as request from @vkuznecovas .
The idea is that all the management is getting too complex, so we want to decouple different components/features by using event bus, signalling all events (service start, stop and possibly others).

At this point, this event doesn't seem very useful, but it's a start towards decoupling :)